detectable/src/main/java/com/synopsys/integration/detectable/detectables/go/gomod/GoModCommandRunner.java

```diff
@@ -0,0 +1,87 @@
+package com.synopsys.integration.detectable.detectables.go.gomod;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.synopsys.integration.detectable.ExecutableTarget;
+import com.synopsys.integration.detectable.ExecutableUtils;
+import com.synopsys.integration.detectable.detectable.executable.DetectableExecutableRunner;
+import com.synopsys.integration.detectable.detectable.executable.ExecutableFailedException;
+
+public class GoModCommandRunner {
+    // java:S5852: Warning about potential DoS risk.
+    @SuppressWarnings({ "java:S5852" })
+    private static final Pattern GENERATE_GO_LIST_JSON_OUTPUT_PATTERN = Pattern.compile("\\d+\\.[\\d.]+"); // Example: "go version go1.17.5 darwin/amd64" -> ""
+    private static final String JSON_OUTPUT_FLAG = "-json";
+    private static final String MODULE_OUTPUT_FLAG = "-m";
+
+    private final DetectableExecutableRunner executableRunner;
+
+    public GoModCommandRunner(DetectableExecutableRunner executableRunner) {
+        this.executableRunner = executableRunner;
+    }
+
+    // Excludes the "all" argument to return the root project modules
+    public List<String> runGoList(File directory, ExecutableTarget goExe) throws ExecutableFailedException {
+        return executableRunner.executeSuccessfully(ExecutableUtils.createFromTarget(directory, goExe, "list", MODULE_OUTPUT_FLAG, JSON_OUTPUT_FLAG))
+            .getStandardOutputAsList();
+    }
+
+    // TODO: Utilize the fields "Main": true, and "Indirect": true, fields from the JSON output to avoid running go list twice. Before switching to json output we needed to run twice. JM-01/2022
+    public List<String> runGoListAll(File directory, ExecutableTarget goExe) throws ExecutableFailedException {
+        boolean readOnlyFlagSupported;
+        try {
+            readOnlyFlagSupported = isReadOnlyFlagSupported(directory, goExe);
+        } catch (ExecutableFailedException e) {
+            // TODO: If we don't have a version for go, we don't do anything. This should probably result in a failure. JM-01/2022
+            return new ArrayList<>();
+        }
```

Comment:
I'm not sure if it would be a considered a breaking change to throw this exception instead. Today we return no results when we can't query `go version` which doesn't seem right to me.

Reply:
I agree it makes more sense to throw error than say nothing and return no results.  I think we have precedent suggesting that it wouldn't be a breaking change (I'm thinking of the ticket where we made a failed go mod why command trigger a failure).

Reply:
I agree, returning an empty BOM makes now sense. The `ExecutableFailedException` will now bubble up.